### ISSUE TITLE
[bug]: During mempool high fees multiple HTLC's are getting stuck pending

### ISSUE THREAD

**Issue by aluam** [lightningnetwork/lnd#7706]
### Background

Recently during mempool high fees I had multiple HLTC's stuck pending on my node with no logs information and zero output from `lncli trackpayment`, this happened multiple times during last two weeks. All peers were online during the whole time.

### Your environment

* version of `lnd` : 0.16.2
* operating system:   Debian Linux 6.1.21-v8+ aarch64
* version of `btcd`, `bitcoind`, or other backend : 24.0.1

### Steps to reproduce

No idea how this happened, no logs or traces of this problem, could not find single information about the forwards being signed in logs. All that I was able to see were multiple HTLC's by using `lncli listchannels | jq '.channels[] | select(.pending_htlcs | length > 0)'` command thanks to @bitromortac

### Expected behaviour

HLTC's should be resolved in time without getting stuck randomly causing force-closes.

### Actual behaviour

HTLC's timed out, some caused force-closure of channel, some were sent on-chain without force close. Latest HTLC's were saved by advice from another node owner by disconnecting the peers that had pending HLTC's with `lncli disconnect pubkey` just about 12 blocks before expiring, this cleared the pending HTLC every time.

**Comment by ffranr** [comment]
We're actively working on this issue. See: https://github.com/lightningnetwork/lnd/issues/7682

**Comment by yyforyongyu** [comment]
What do you mean that "HTLC is stuck"? Is the outgoing HTLC tx stuck in mempool?

> HTLC's timed out, some caused force-closure of channel, some were sent on-chain without force close.

If it went onchain then the channel must have been force closed.


> Latest HTLC's were saved by advice from another node owner by disconnecting the peers that had pending HLTC's with lncli disconnect pubkey just about 12 blocks before expiring, this cleared the pending HTLC every time.

What's the pending HTLC?

**Comment by aluam** [comment]
> What do you mean that "HTLC is stuck"? Is the outgoing HTLC tx stuck in mempool?
> 
Not stuck in mempool, those forwards were pending in LND without any attempts to resolve.
> 
> If it went onchain then the channel must have been force closed.
> 
One of the HLTC expired and did not cause force-close, or maybe it was just coincidence of another HTLC being resolved from other closures, not sure. It was quite messy on my node after all the force-closures and sweeps I had.
> 
> What's the pending HTLC?

You should probably have a look at the issue mentioned above, which will explain it better than me. I was told to create new one here, just not good at wording.

**Comment by yyforyongyu** [comment]
> Not stuck in mempool, those forwards were pending in LND without any attempts to resolve.

Ok then in theory this has nothing to do with high fees. Could you search your logs to find the line `failing link: unable to complete dance with error`? And maybe share some debug logs around that period where you saw it's stuck, that'd be very helpful, thanks!

**Comment by Roasbeef** [comment]
@aluam you mentioned a reconnect cycle worked, do you know if that was a Tor peer?

**Comment by aluam** [comment]
@yyforyongyu I was searching through the logs up to one week past and there is zero mention about : `"failing link: unable to complete dance"` 

There is only one mention of the channel in question with other failing link message: 
`2023-05-15 15:57:42.496 failing link: unable to synchronize channel states: unable to send chan sync message for ChannelPoint(txid:output) write tcp : write: broken pipe with error: unable to resume channel, recovery required` which was about day before I had those HTLC's pending. But after that its the usual "`received re-establishment`" message.  


This part of log is during the manual peer disconnect before the HTLC expired. Doesn't look like anything out of the ordinary.
```
2023-05-16 15:01:42.504 [INF] HSWC: ChannelLink(): stopping
2023-05-16 15:01:42.504 [INF] HSWC: ChannelLink(): exited
2023-05-16 15:01:44.245 [INF] PEER: Peer(): Loading ChannelPoint(), isPending=false
2023-05-16 15:01:44.246 [INF] HSWC: ChannelLink(): starting
2023-05-16 15:01:44.247 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint()
2023-05-16 15:01:44.247 [INF] HSWC: ChannelLink(): HTLC manager started, bandwidth= mSAT
2023-05-16 15:01:44.247 [INF] HSWC: ChannelLink(): attempting to re-synchronize
2023-05-16 15:01:44.429 [INF] HSWC: ChannelLink(): received re-establishment message from remote side

```


@Roasbeef Actually yes, all of the peers that I had issues with recently are Tor-only nodes, did not realize that. All the force closes are related to Tor users as well.

**Comment by saubyk** [comment]
@Roasbeef  @yyforyongyu 

Does this pr https://github.com/lightningnetwork/lnd/pull/7711 address this issue?

**Comment by yyforyongyu** [comment]
@saubyk This is different from that issue since there's no stuck commit dance, which is identified by the line `failing link: unable to complete dance with error: remote unresponsive`.

**Comment by SunnySarahNode** [comment]
It looks like the author means the following.

Normal process (low onchain fees environment) :

(1) There is some HTLC
(2) Some channel on its route was FCed
(3) The closing transaction was mined quickly
(4) All relevant HTLCs were failed
(5) HTLC (1) was deleted

High onchain fees environment :

(1) There is some HTLC
(2) Some channel on its route was FCed
(3) The closing transaction was NOT mined due to low default `max-commit-fee-rate-anchors` value, that was described in #8240 ( example tx : `ce960ba459e62fbbe9178130de89fb595afa8ffb390b954d3e3f3aaf4e0f3f56` )
(4) All relevant HTLCs including (1) remain alive and receive the "pending" status ( in `lncli listchannels` )
(5) This causes a cascade of FCes, that was described in detail in #7683 and #7969

Looks like this issue should be closed as a duplicate of #7683

**Comment by Roasbeef** [comment]
Closing as a duplicate of: https://github.com/lightningnetwork/lnd/issues/7683
